processor/transform/tables-to-jsx.ts

```diff
@@ -0,0 +1,98 @@
+import { Parents, Table, TableCell, Text } from 'mdast';
+import { visit, EXIT } from 'unist-util-visit';
+import { Transform } from 'mdast-util-from-markdown';
+
+import { mdxJsx } from 'micromark-extension-mdx-jsx';
+import { fromMarkdown } from 'mdast-util-from-markdown';
+import { mdxJsxFromMarkdown } from 'mdast-util-mdx-jsx';
+import { phrasing } from 'mdast-util-phrasing';
+
+const visitor = (table: Table, index: number, parent: Parents) => {
+  let hasFlowContent = false;
+
+  const tableCellVisitor = (cell: TableCell) => {
+    if (!phrasing(cell.children[0])) {
+      hasFlowContent = true;
+      return EXIT;
+    }
+
+    visit(cell, 'text', (text: Text) => {
+      if (text.value.match(/\n/)) {
+        hasFlowContent = true;
+        return EXIT;
+      }
+    });
+  };
+
+  visit(table, 'tableCell', tableCellVisitor);
+  if (!hasFlowContent) return;
+
+  const head = {
+    type: 'mdxJsxFlowElement',
+    name: 'thead',
+    children: [
+      {
+        type: 'mdxJsxFlowElement',
+        name: 'tr',
+        children: table.children[0].children.map((cell, idx) => {
+          const proxy = fromMarkdown(`<div style={{ align: "${table.align[idx]}" }}></div>`, {
+            extensions: [mdxJsx()],
+            mdastExtensions: [mdxJsxFromMarkdown()],
+          });
+          // @ts-ignore
+          const { attributes } = proxy.children[0];
+
+          return {
+            type: 'mdxJsxFlowElement',
+            name: 'th',
+            attributes,
+            children: cell.children,
+          };
+        }),
+      },
+    ],
+  };
+
+  const body = {
+    type: 'mdxJsxFlowElement',
+    name: 'tbody',
+    children: table.children.splice(1).map(row => {
+      return {
```

Comment:
Could simplify, if you're so inclined!

```suggestion
    children: table.children.splice(1).map(row => ({
```

Reply:
Missed this before merging, oops!

Reply:
ehh, is the least important suggestion of all time!